### ISSUE TITLE
needless_range_loop false positive: loop variable *not* used to index sequence

### ISSUE THREAD

**Issue by vks** [Manishearth/rust-clippy#1803]
```rust
for mid in 1..sequence.len() {
    let (left, right) = sequence.split_at(mid);
    // ...
}
```
results in
```
warning: the loop variable `mid` is used to index `sequence`
[...]
help: consider using an iterator
   |     for (mid, <item>) in sequence.iter().enumerate().skip(1) {
```

This suggestion does not make sense, because we don't even need the item, we are only interested in the index for calling `split_at`.
